lib/Mojo/mysql/PubSub.pm

```diff
@@ -100,14 +131,57 @@ sub _wait_db {
     my $sync_db = $self->mysql->db;
     return (delete $self->{wait_db}, $self->_wait_db($sync_db)) if $err;
     $res->finish;
-    $db->query('select sleep(600)', $cb);
-    $sync_db->query('update mojo_pubsub_subscribe set ts = current_timestamp where pid = ?', $db->pid);
+    _query_with_retry(
+      $db,
+      'select sleep(600)',
+      $cb
+    );
+    _query_with_retry(
+      $sync_db,
+      'update mojo_pubsub_subscribe set ts = current_timestamp where pid = ?',
+      $db->pid
+    );
     $self->_notifications($self->mysql->db);
   };
 
   warn qq|[PubSub] (@{[$wait_db->pid]}) reconnect\n| if DEBUG;
   $self->emit(reconnect => $wait_db);
-  return $wait_db->query('select sleep(600)', $cb);
+  return _query_with_retry(
+    $wait_db,
+    'select sleep(600)',
+    $cb
+  );
+}
+
+sub _query_with_retry {
+  my ($db, @query_args) = @_;
+
+  my $wantarray = wantarray;
```

Comment:
Why is `wantarray` added? I would appreciate if it was removed.

Reply:
Will do - I'm used to working with DBIx::Class where it can return different stuff depending on calling context, but looking at `Mojo::mysql::Database::query` I see where it only ever returns a single scalar.